spec/rails_helper.rb

```diff
@@ -810,6 +810,15 @@ def file_from_fixtures(filename, directory = "images")
   File.new(tmp_file_path)
 end
 
+def plugin_file_from_fixtures(plugin_directory, filename, directory = "images")
+  tmp_file_path = File.join(concurrency_safe_tmp_dir, SecureRandom.hex << filename)
+  FileUtils.cp(
+    "#{Rails.root}/plugins/#{plugin_directory}/spec/fixtures/#{directory}/#{filename}",
+    tmp_file_path,
+  )
+  File.new(tmp_file_path)
+end
```

Comment:
```suggestion
def file_from_fixtures(filename, directory = "images", root_path: "#{Rails.root}/spec/fixtures")
  tmp_file_path = File.join(concurrency_safe_tmp_dir, SecureRandom.hex << filename)
  FileUtils.cp("#{root_path}/#{directory}/#{filename}", tmp_file_path)
  File.new(tmp_file_path)
end

def plugin_file_from_fixtures(plugin_directory, filename, directory = "images")
  file_from_fixtures(filename, directory, root_path: "#{Rails.root}/plugins/#{plugin_directory}/spec/fixtures")
end
```

We can tweak `file_from_fixtures` alittle so avoid duplicating the concurrency stuff.

Reply:
Hah yeah fair enough, I had considered this too. Will change :+1: